Integrations/integration-Jask.yml

```diff
@@ -0,0 +1,1003 @@
+        """
+        Convert a given string with MM/DD/YYYY format to millis since epoch
+        """
+        d = datetime.strptime(dstr, '%m/%d/%Y')
+        return int((d - datetime.utcfromtimestamp(0)).total_seconds() * 1000)
+
+
+    def getInsightDetails():
+        """
+        Get insight details
+        """
+        alertId = demisto.getArg('insight-id')
+        respJSON = req('GET', 'alert/' + alertId)
+        ec = insight_to_readable(respJSON)
+
+        detailsMD = tableToMarkdown('Insight Details:', [ec], ['Id', 'Name','Action','AssignedTo','Description','IpAddress','LastUpdated','LastUpdatedBy','Severity','InsightTime','WorkflowStatus'])
+
+        entityDisplay = entity_to_readable(respJSON.get('asset_details'))
+        entityDisplay.update({'IpAddress': demisto.get(respJSON['asset_details']['ip'], 'address')})
+        ec['EntityDetails'] = entityDisplay
+        entityDisplay['Id'] = respJSON.get('id')
+
+        entityMarkdown = tableToMarkdown('Insight\'s Main Entity Details:', [entityDisplay], ['Id', 'EntityType', 'Hostname', 'Groups', 'FirstSeen', 'LastSeen', 'IpAddress', 'IsWhitelisted', 'RiskScore', 'Source'])
+
+        relatedAssetsJson = respJSON.get('related_assets') or []
+        resultsAssetsList = []
+        for relAsset in relatedAssetsJson:
+            resultsAsset = to_readable(relAsset, ['id', 'is_whitelisted', 'risk_score', 'source', 'asset_type'], {'asset_type': 'EntityType'})
+            if relAsset.get('asset_type') == 'hostname':
+                resultsAsset['Name'] = relAsset.get('hostname')
+            elif relAsset.get('asset_type') == 'username':
+                resultsAsset['Name'] = relAsset.get('username')
+
+            if demisto.get(relAsset, 'ip.address'):
+                resultsAsset['IpAddress'] = demisto.get(relAsset, 'ip.address')
+
+            resultsAssetsList.append(resultsAsset)
+        ec['RelatedEntityList'] = resultsAssetsList
+        relAssetsMD = tableToMarkdown('Related Entities:', resultsAssetsList, ['Id', 'EntityType', 'IpAddress', 'Name', 'IsWhitelisted', 'RiskScore', 'Source'])
+
+        signalListJson = respJSON.get('signals') or []
+        signalList = []
+        threatIntel = 0
+        anomalies = 0
+        patterns = 0
+        for signalItem in signalListJson:
+            resultSignal = signal_to_readable(signalItem)
+            if resultSignal['SourceType'] == 'threatintel':
+                threatIntel += 1
+            elif resultSignal['SourceType'] == 'rule':
+                patterns += 1
+            elif resultSignal['SourceType'] == 'anomaly':
+                anomalies += 1
+            signalList.append(resultSignal)
+        ec['SignalList'] = signalList
+        signalsMD = tableToMarkdown('Related Signals:', signalList, ['Id', 'Name', 'Description', 'Category', 'SourceType'])
+        ec['SignalListMetadata'] = {'Patterns': {'Count': patterns}, 'Anomalies': {'Count': anomalies}, 'ThreatIntel': {'Count': threatIntel}}
```

Comment:
don't write one line dictionary/json, instead format all the dictionaries with indentation.
It is more readable 
```
{
    'Patterns': {
        'Count': patterns
    }, 
    'Anomalies': {
        'Count': anomalies
    }, 
    'ThreatIntel': {
        'Count': threatIntel
    }
}
```